src/reader/metadata.rs

```diff
@@ -22,153 +47,234 @@ pub struct FileMetadata {
     pub stripe_footers: Vec<StripeFooter>,
 }
 
-macro_rules! impl_read_metadata {
-    ($reader:ident $($_await:tt)*) => {
-        {
```

Comment:
I removed the macro as I find it hard to develop with (using Helix + Rust-analyzer, my tooling doesn't play nice with it, hard to refactor)

Since at most its for two functions (sync and async) I think its fine to duplicate